x-pack/test/api_integration/deployment_agnostic/apis/observability/ai_assistant/complete/functions/execute_query.spec.ts

```diff
@@ -0,0 +1,210 @@
+          name: 'query',
+          arguments: () => JSON.stringify({}),
+          when: () => true,
+        });
+
+        void llmProxy.interceptWithFunctionRequest({
+          name: 'structuredOutput',
+          arguments: () => JSON.stringify({}),
+          // @ts-expect-error
+          when: (requestBody) => requestBody.tool_choice?.function?.name === 'structuredOutput',
+        });
+
+        void llmProxy.interceptWithFunctionRequest({
+          name: 'execute_query',
+          arguments: () =>
+            JSON.stringify({
+              query: `FROM logs-apache.access-default
+            | KEEP message
+            | SORT @timestamp DESC
+            | LIMIT 10`,
+            }),
+          when: () => true,
+        });
+
+        void llmProxy.interceptConversation({ content: 'Hello from user' });
+
+        ({ messageAddedEvents } = await chatComplete({
+          userPrompt: 'Please retrieve the most recent Apache log messages',
+          connectorId,
+          observabilityAIAssistantAPIClient,
+        }));
+
+        await llmProxy.waitForAllInterceptorsToHaveBeenCalled();
+        firstRequestBody = llmProxy.interceptedRequests[0].requestBody;
+        secondRequestBody = llmProxy.interceptedRequests[1].requestBody;
+        thirdRequestBody = llmProxy.interceptedRequests[2].requestBody;
+        fourthRequestBody = llmProxy.interceptedRequests[3].requestBody;
+      });
+
+      after(async () => {
+        await logsSynthtraceEsClient.clean();
+      });
+
+      it('makes 4 requests to the LLM', () => {
+        expect(llmProxy.interceptedRequests.length).to.be(4);
+      });
+
+      it('emits 7 messageAdded events', () => {
+        expect(messageAddedEvents.length).to.be(7);
+      });
+
+      describe('The first request', () => {
+        it('exposes the right tools', () => {
+          expect(firstRequestBody.tools?.map((t) => t.function.name)).to.eql([
+            'query',
+            'get_alerts_dataset_info',
+            'alerts',
+            'changes',
+            'elasticsearch',
+            'kibana',
+            'get_dataset_info',
+            'execute_connector',
+          ]);
+        });
+      });
+
+      describe('The second request', () => {
+        it('contains the correct number of messages', () => {
+          expect(secondRequestBody.messages.length).to.be(6);
+        });
+
+        it('contains the `structuredOutput` tool choice', () => {
+          // @ts-expect-error
+          const hasToolChoice = secondRequestBody.tool_choice.function?.name === 'structuredOutput';
+          expect(hasToolChoice).to.be(true);
+        });
+
+        it('contains user message with information about how to request ESQL documentation', () => {
+          expect(last(secondRequestBody.messages)?.content).to.contain(
+            'Based on the previous conversation, request documentation'
+          );
+        });
+      });
+
+      describe('The third request', () => {
```

Comment:
(just a suggestion, can be ignored if it doesn't add much value)
```suggestion
      describe('The third request - Requesting ESQL documentation', () => {
```